xc/device.ml

```diff
@@ -2284,21 +2290,25 @@ module Backend = struct
             try
               Monitor.wait m |> Monitor.with_event m (fun fd is_flag_in ->
                   Lookup.domid_of fd >>= fun domid ->
+                  Lookup.channel_of domid >>= fun c ->
+                  let qmp = Qmp_protocol.to_fd c in  
                   if is_flag_in then
-                    Lookup.channel_of domid >>= fun c ->
-                    try
-                      match Qmp_protocol.read c with
-                      | Event e -> qmp_event_handle domid e
-                      | msg -> debug "Got non-event message, domain-%d: %s" domid (string_of_message msg)
-                    with End_of_file ->
+                    match Readln.read qmp with
```

Comment:
This is not Thread safe, but it is fine because qemu blocks when communicates, right?